### ISSUE TITLE
clang: pack expansion contains parameter pack 'Args' that has a different length (1 vs. 2) from outer parameter packs

### ISSUE THREAD

**Issue by jcelerier** [llvm/llvm-project#64902]
Hello,

repro:

```c++
#include <functional>

struct worker {
    std::function<void(int, int)> request;
    static void work(int, int);
} worker;

template <typename T = int>
void connect_worker() {
    worker.request = []<typename... Args>(Args&&... f) {
        [... ff = std::forward<Args>(f)]() mutable {
            worker::work(std::forward<Args>(ff)...);
        }();
    };
}

int main() { connect_worker(); }
```

causes: 

```console
<source>:14:48: error: pack expansion contains parameter pack 'Args' that has a different length (1 vs. 2) from outer parameter packs
            worker::work(std::forward<Args>(ff)...);
                                      ~~~~     ^
```

I'm confident this is a compiler bug as if one turns `connect_worker` into a non-template by removing `
template <typename T = int>`, then the code builds and runs as expected:

https://gcc.godbolt.org/z/YM6ar7WG3

also,

```cpp
static_assert(sizeof...(Args) == 2);
```

Just before the offending line passes too.

Not forwarding the arguments works too: 

```cpp
worker::work(ff...);
```

**Comment by shafik** [comment]
Maybe duplicate of: https://github.com/llvm/llvm-project/issues/61415

CC @erichkeane

**Comment by zyn0217** [comment]
Should have been fixed by #101385
